frontend/lib/src/components/elements/DeckGlJsonChart/withMapboxToken/withMapboxToken.tsx

```diff
@@ -71,17 +85,48 @@ const withMapboxToken =
           mapboxToken: undefined,
           mapboxTokenError: undefined,
         }
+      }
+
+      /**
+       * Expose a singleton MapboxToken:
+       * - If the user specified a token in their streamlit config, return it.
+       * - Else, fetch the remote "tokens.json" and return the "mapbox" entry.
+       *
+       * (The returned value is cached in memory, so the remote resource will
+       * only be fetched once per session.)
+       */
+      public async getMapboxToken(sessionInfo: SessionInfo): Promise<string> {
+        let { userMapboxToken } = sessionInfo.current
+
+        if (userMapboxToken) {
+          return userMapboxToken
+        }
+
+        try {
+          const response = await axios.get(TOKENS_URL)
+          const { [MAPBOX]: token } = response.data
+
+          if (!token) {
+            throw new Error(`Missing token ${MAPBOX}`)
+          }
+
+          userMapboxToken = token
+        } catch (e) {
+          const error = ensureError(e)
+          throw new MapboxTokenFetchingError(
+            `${error.message} (${TOKENS_URL})`
+          )
+        }
 
-        this.initMapboxToken()
+        return userMapboxToken
       }
 
       /**
        * Fetch our MapboxToken.
        */
       public initMapboxToken = async (): Promise<void> => {
         try {
-          const mapboxToken = await MapboxToken.get(this.props.sessionInfo)
-
+          const mapboxToken = await this.getMapboxToken(this.props.sessionInfo)
```

Comment:
I read in the streamlitview slack channel that components require `props. sessionInfo` are not supported. The DeckGLJsonChart is the only component that requires the session info:

https://github.com/streamlit/streamlit/blob/8b1eee9bb0815f86077e71161556e1a0811df422/frontend/lib/src/components/core/Block/ElementNodeRenderer.tsx#L321 

In general, it could be beneficial to try to refactor this out. One potential way of doing that is to add a mapboxToken property to the `DeckGlJsonChart.proto` that gets filled in the pydeck marshall command with the token from the config. Thereby, we can also deprecate/remove this token from the session info.

Reply:
Moving this token property to the DeckGlJsonChart.proto would probably make things cleaner... especially since no other component is using the token.

Reply:
Hm I looked into other things that use mapbox tokens and that would be plotly. However, it doesn't seem like any issues pop up for plotly and mapbox maybe besides this one ?

https://github.com/streamlit/streamlit/issues/6121

As a result, will go through and refactor this.